core/src/test/java/org/everit/json/schema/loader/SchemaLoaderTest.java

```diff
@@ -484,4 +467,30 @@ public void relativeIdInReferencedSchemaRoot() {
                 .load().build();
     }
 
+    @Test
+    public void nullableTest() {
+        JSONObject rawSchema = ALL_SCHEMAS.getJSONObject("nullableTest");
+        ObjectSchema schema = (ObjectSchema) SchemaLoader.builder().schemaJson(rawSchema).OAS3SchemaSupport().build().load().build();
+
+        try {
+            schema.validate(new JSONObject("{\"hello\": null, \"world\": \"yee\"}"));
+            assertEquals(true, true);
+        } catch (ValidationException e) {
+            assertTrue("Thrown unexpected exception " + e, false);
+        }
+    }
+
+    @Test
+    public void nullableTestError() {
+        JSONObject rawSchema = ALL_SCHEMAS.getJSONObject("nullableTest");
+        ObjectSchema schema = (ObjectSchema) SchemaLoader.builder().schemaJson(rawSchema).OAS3SchemaSupport().build().load().build();
+
+        try {
+            schema.validate(new JSONObject("{\"hello\": \"world\", \"world\": null}"));
+            assertTrue("Exception not thrown!", false);
+        } catch (ValidationException e) {
+            assertEquals(true, true);
+        }
+    }
```

Comment:
The convention is that SchemaLoader unittests only verify that the proper Schema object is constructed from the raw json, so in this case, in the  `SchemaLoaderTest` you should just make assertions on the `getNullable()` flag of the created Schema instance.

Then to check if the `nullable` flag is properly handled by the Schema subclasses, you can add a new testcase for each `Schema` subclass to check if the nullable flag is taken into account properly (that means one check for each `checkNullity()` call). For these tests, please use the `TestSupport` class which helps you to expect conventional `ValidationException` instances (which is a bit more detailed than exceptions usually are).